### PR TITLE
fix: make `importModels` support non-default CJS exports

### LINKED ISSUE #16843: importModels does not work as expected
## Issue Creation Checklist

- [x] I understand that my issue will be automatically closed if I don't fill in the requested information
- [x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

### Reproducible Example

https://github.com/ckvv/sequelize-import

Here is the link to the SSCCE for this issue:

### What do you expect to happen?

[importModels](https://github.com/sequelize/sequelize/blob/main/packages/core/src/import-models.ts#L16-L34) does not work as expected

src/mock/index.ts

```ts
import { join } from 'node:path';
import { importModels } from '@sequelize/core';

(async () => {
  console.log(await importModels(join(__dirname, '../models/*.model.ts')))
})()
```

src/models/m1.model.ts
```ts
import type {
  InferAttributes,
  InferCreationAttributes,
} from '@sequelize/core';
import {
  DataTypes,
  Model,
} from '@sequelize/core';
import {
  Attribute,
} from '@sequelize/core/decorators-legacy';

export class M1 extends Model<
  InferAttributes<M1>,
  InferCreationAttributes<M1>
> {
  @Attribute(DataTypes.INTEGER)
  declare id: number;
}

```
src/models/m2.model.ts

```ts
import type {
  InferAttributes,
  InferCreationAttributes,
} from '@sequelize/core';
import {
  DataTypes,
  Model,
} from '@sequelize/core';
import {
  Attribute,
} from '@sequelize/core/decorators-legacy';

export class M2 extends Model<
  InferAttributes<M2>,
  InferCreationAttributes<M2>
> {
  @Attribute(DataTypes.INTEGER)
  declare id: number;
}
```

run `npx tsx ./src/mock/index.ts`

expect `[ M1, M2 ]`
Print `[]`

### What is actually happening?

It was working fine before, The code has not changed, I'm not sure what I did to cause the `import()` value to always be wrapped by `{ default : ...}`

edit file `node_modules/@sequelize/core/lib/import-models.js`

```js
async function importModelNoGlob(path, modelMatch) {
  const module2 = await import(path);

  console.log(module2) // add console
  return Object.keys(module2).filter((exportName) => {
    if (!(0, import_model_utils.isModelStatic)(module2[exportName])) {
      return false;
    }
    if (modelMatch) {
      return modelMatch(path, exportName, module2[exportName]);
    }
    return true;
  }).map((exportName) => module2[exportName]);
}
```

Print 
```
[Module: null prototype] { default: { M1: [Getter] } }
[Module: null prototype] { default: { M2: [Getter] } }
```

expect
```
{ M1: [Getter] } }
{ M2: [Getter] } }
```

### Environment

- Sequelize version:7.0.0-alpha.29
- Node.js version:v20.10.0
- If TypeScript related: TypeScript version:
- Database & Version:
- Connector library & Version:

## Would you be willing to resolve this issue by submitting a Pull Request?

- [x] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I will need guidance.
- [ ] No, I don't have the time, but my company or I are [supporting Sequelize through donations on OpenCollective](https://opencollective.com/sequelize).
- [ ] No, I don't have the time, and I understand that I will need to wait until someone from the community or maintainers is interested in resolving my issue.

---

*Indicate your interest in the resolution of this issue by adding the 👍 reaction. Comments such as "+1" will be removed.*

### PR DESCRIPTION
**Pull request by ckvv**
Fixes #16843

### CODE REVIEW

File: packages/core/src/import-models.ts [hunk 1]
```diff
@@ -35,6 +35,9 @@ export async function importModels(globPaths: string | string[], modelMatch?: Mo
 
 async function importModelNoGlob(url: string, modelMatch?: ModelMatch): Promise<ModelStatic[]> {
   const module = await import(url);
+  if (module.default && !isModelStatic(module.default) && typeof module.default === 'object') {
+    Object.assign(module, module.default);
```
Comment (ephys) [inline packages/core/src/import-models.ts:39]: modules are typically sealed, so you need to create a copy like below:

image

```suggestion
    module = { ...module, ...module.default };
```

File: packages/core/src/import-models.ts [hunk 2]
```diff
@@ -35,6 +35,9 @@ export async function importModels(globPaths: string | string[], modelMatch?: Mo
 
 async function importModelNoGlob(url: string, modelMatch?: ModelMatch): Promise<ModelStatic[]> {
   const module = await import(url);
+  if (module.default && !isModelStatic(module.default) && typeof module.default === 'object') {
```
Comment (ephys) [inline packages/core/src/import-models.ts:38]: Adding some info for future maintainers

```suggestion
  // when importing a CJS file, sometimes only the default export is available, as named exports depend on the file's exports being statically analyzable by node. The default exports contains the contents of the file's `module.exports`
  if (module.default && !isModelStatic(module.default) && typeof module.default === 'object') {
```

**Comment by ephys** [comment]
Could you add a unit test that tries to import a CJS file with "named" exports? This should cover what is added in this PR

You can do that in `import-models.test.ts`

**Review by ephys — CHANGES_REQUESTED** [review]

**Comment by ephys** [comment]
Made a tiny change to only spread plain objects (objects with either no prototype, or `Object.prototype` as the prototype). If the default export is a class, or a date, etc, we should not spread it

**Review by WikiRik — APPROVED** [review]

**Pull request merged**